h2o-py/tests/testdir_sklearn/pyunit_sklearn_automl_pipeline.py

```diff
@@ -0,0 +1,210 @@
+from __future__ import print_function
```

Comment:
@ledell This test file shows how it would work with AutoML.
The pipelines are simpler than than with `h2o-py/tests/testdir_sklearn/pyunit_sklearn_classification_pipeline.py` for example (didn't use transformers here for simplicity), but more complex pipelines or even sklearn searches would work as well here as for other H2O estimators.
In those tests, I just verified the various use cases:
- connection handling (autostart).
- data input format: H2OFrame vs. numpy arrays.
- setting parameters the sklearn way: constructor, set_params, as properties.
- sklearn API: `fit`, `predict`, `predict_proba`, `score`

Reply:
@seb-h2o Do this support `pandas.DataFrame` in addition to numpy arrays?

Reply:
@ledell for data input yes, we can use `pandas.DataFrame`, but the auto-converted output for `predict` and `transform` will still be `numpy` arrays for now.
I need to double check what `sklearn` does currently: my first idea was do support `pandas` input -> `pandas` output, but I think that even `sklearn` does `pandas` input -> `numpy` output, that's why I removed conversion to `pandas`: should not be difficult to reintroduce though.

Reply:
I confirm that `sklearn` itself doesn't return `pandas.DataFrame` on `predict` and `transform` methods, so I don't need to support back-conversion to pandas.
However, it looks like my conversion from `pandas.DataFrame` as input data is broken (was relying on H2OFrame support for pandas but header is handled correctly only on some conditions): will fix this and add a test for `pandas`.

Reply:
pandas support fixed: it was a corner case in our H2OFrame pandas support when pandas DataFrame column names are not strings (e.g. when dataframe is directly created from numpy array)